src/clj/rems/poller/email.clj

```diff
@@ -0,0 +1,225 @@
+(ns rems.poller.email
+  "Sending emails based on application events."
+  (:require [clojure.test :refer :all]
+            [clojure.tools.logging :as log]
+            [mount.core :as mount]
+            [postal.core :as postal]
+            [rems.config :refer [env]]
+            [rems.db.applications :as applications]
+            [rems.db.core :as db]
+            [rems.db.users :as users]
+            [rems.text :as text]
+            [rems.json :as json]
+            [rems.util :as util]
+            [rems.workflow.dynamic :as dynamic]))
+
+;;; Mapping events to emails
+
+;; TODO link to application?
+;; TODO list of resources?
+;; TODO use real name when addressing user?
+
+(defmulti ^:private event-to-emails-impl
+  (fn [event _application] (:event/type event)))
+
+(defmethod event-to-emails-impl :default [_event _application]
+  [])
+
+(defmethod event-to-emails-impl :application.event/approved [event application]
+  (vec
+   (for [member (:members application)] ;; applicant is a member
+     {:to-user (:userid member)
+      :subject (text/text :t.email/application-approved-subject)
+      :body (text/text-format :t.email/application-approved-message
+                              (:userid member)
+                              (:id application))})))
+
+(defmethod event-to-emails-impl :application.event/rejected [event application]
+  (vec
+   (for [member (:members application)] ;; applicant is a member
+     {:to-user (:userid member)
+      :subject (text/text :t.email/application-rejected-subject)
+      :body (text/text-format :t.email/application-rejected-message
+                              (:userid member)
+                              (:id application))})))
+
+(defmethod event-to-emails-impl :application.event/comment-requested [event _application]
+  (vec
+   (for [c (:application/commenters event)]
```

Comment:
Longer variable name (`c` -> `commenter`) would be nice. Also in the methods below this one.

Reply:
done